src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -439,10 +484,19 @@ private static void AddGrpcAttributes(Activity activity, string grpcMethod, Http
             activity.SetTag(SemanticConventions.AttributeRpcSystem, GrpcTagHelper.RpcSystemGrpc);
             if (context.Connection.RemoteIpAddress != null)
             {
+                // TODO: This attribute was changed in v1.13.0 https://github.com/open-telemetry/opentelemetry-specification/pull/2614
                 activity.SetTag(SemanticConventions.AttributeNetPeerIp, context.Connection.RemoteIpAddress.ToString());
             }
 
-            activity.SetTag(SemanticConventions.AttributeNetPeerPort, context.Connection.RemotePort);
+            if (this.httpSemanticConvention.HasFlag(HttpSemanticConvention.Old))
+            {
+                activity.SetTag(SemanticConventions.AttributeNetPeerPort, context.Connection.RemotePort);
+            }
+
+            if (this.httpSemanticConvention.HasFlag(HttpSemanticConvention.New))
+            {
+                activity.SetTag(SemanticConventions.AttributeClientSocketPort, context.Connection.RemotePort);
```

Comment:
Are we clear on what this tag is meant for? And what does this value `context.Connection.RemotePort` represent?

Reply:
I reviewed the spec and confirmed this was the wrong attribute. will change to `server.port`

- v1.20.0
  https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#http-client
  ![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/28785781/9fc3610d-3bc1-4f57-811c-254159764c1f)

- v1.21.0
  https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions
  ![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/28785781/5e2c5783-04d5-454d-a2cb-ae8897fe3825)

Reply:
Regarding the value of `context.Connection.RemotePort`, the documentation states:
> Gets or sets the port of the remote target
https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.connectioninfo.remoteport?view=aspnetcore-7.0

This is not changed, it was mapped to the former attribute.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/2a6d88c35749c181f5c0816e994965c88228af49/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L491-L494